src/mat.c

```diff
@@ -883,7 +885,8 @@ Mat_VarDelete(mat_t *mat, const char *name)
                 break;
         }
 
-        tmp = Mat_CreateVer(tmp_name,mat->header,mat_file_ver);
+        close(fd);
+        tmp = Mat_CreateVer(temp,mat->header,mat_file_ver);
```

Comment:
This introduces a race condition: After closing the file descriptor `fd` of the temporary file `temp` might no longer be a unique temporary filename. Thus a proper fix also requires replacing all `fwrite` calls by ``write` (which is some effort).